crates/core/src/network/connection_manager.rs

```diff
@@ -271,6 +304,20 @@ impl<T: TransportConnect> ConnectionManager<T> {
         if let Some(connection) = maybe_connection {
             return Ok(connection);
         }
+
+        let is_gone = {
+            self.inner
+                .lock()
+                .observed_generations
+                .get(&node_id.as_plain())
+                .map(|status| node_id.generation() <= status.generation && status.gone)
```

Comment:
Could we make this filter even a bit more selective by `node_id.generation() < status.generation || node_id.generation() == status.generation && status.gone`?

Reply:
Alrighty.